indico/modules/users/controllers.py

```diff
@@ -306,40 +306,39 @@ def _process(self):
 
 class RHUserFavorites(RHUserBase):
     def _process(self):
-        query = (Category.query
-                 .filter(Category.id.in_(c.id for c in self.user.favorite_categories))
-                 .options(undefer('chain_titles')))
-        categories = sorted([(cat, truncate_path(cat.chain_titles[:-1], chars=50)) for cat in query],
-                            key=lambda c: (c[0].title, c[1]))
-        return WPUser.render_template('favorites.html', 'favorites', user=self.user, favorite_categories=categories)
+        return WPUserFavorites.render_template('favorites.html', 'favorites', user=self.user)
 
 
-class RHUserFavoritesUsersAdd(RHUserBase):
-    def _process(self):
-        users = [User.get(int(id_)) for id_ in request.form.getlist('user_id')]
-        self.user.favorite_users |= {_f for _f in users if _f}
-        tpl = get_template_module('users/_favorites.html')
-        return jsonify(success=True, users=[serialize_user(user) for user in users],
-                       html=tpl.favorite_users_list(self.user))
+class RHUserFavoritesAPI(RHProtected):
+    def _process_args(self):
+        self.user = User.get_or_404(request.view_args['fav_user_id']) if 'fav_user_id' in request.view_args else None
```

Comment:
Careful, this exposes everyone's favorites to any other user. Probably better to just inherit from `RHUserBase` which already assigns `self.user` properly and performs an access check when specifying a user that is not you.